File: ExamenMoria_Java/src/main/java/models/item/Anillo.java
package models.item;

import interfaces.Hobbit;

import java.util.UUID;

public class Anillo extends Item implements Hobbit {

    public boolean llevaElAnillo = false;

    public Anillo(UUID id) {
        super(id);
    }

    @Override
    public String toString() {
        return "Anillo{" +
                "llevaElAnillo=" + llevaElAnillo +
                ", id=" + id +
                '}';
    }

    public boolean elLlevaElAnillo() {
        return llevaElAnillo;
    }

    @Override
    public void ponerAnillo() {
        System.out.println("Tratamos de ponernos el anillo.");

        double randomValue = Math.random();

        if(randomValue <= 0.5){
            System.out.println("Hemos conseguido ponerlos el anillo.");
            llevaElAnillo = true;
        }else{
            quitarseAnillo();
        }
    }

    @Override
    public void quitarseAnillo() {
        System.out.println("No hemos conseguido ponerlos el anillo.");
        llevaElAnillo = false;
    }
}

File: ExamenMoria-Python/projecto/models/items/Anillo.py
from Item import Item

from interfaz.Hobbit import Hobbit
import random

class Anillo(Item, Hobbit):
    def __init__(self, id: int, estaPuesto: bool):
        super().__init__(id)
        self.estaPuesto = estaPuesto

    def __str__(self) -> str:
        return f"Anillo({self.id}, {self.estaPuesto})"

    def ponerseElAnillo(self):
        print("Tratamos de ponernos el anillo.")

        rng = random.randint(1,100)

        if rng <= 50:
            print("Hemos conseguido ponernos el anillo.")
            self.estaPuesto = True
        else:
            self.quitarseElAnillo()
    
    def quitarseElAnillo(self):
        print("No hemos conseguido ponernos el anillo.")
        self.estaPuesto = False